backend/handler/filesystem/roms_handler.py

```diff
@@ -84,12 +86,39 @@ def _exclude_multi_roms(self, roms) -> list[str]:
 
         return [f for f in roms if f not in filtered_files]
 
+    def _calculate_rom_size(self, data: bytes) -> int:
+        return {
+            "crc_hash": (binascii.crc32(data) & 0xFFFFFFFF)
+            .to_bytes(4, byteorder="big")
+            .hex(),
+            "md5_hash": hashlib.md5(data).hexdigest(),
+            "sha1_hash": hashlib.sha1(data).hexdigest(),
+        }
+
     def get_rom_files(self, rom: str, roms_path: str) -> list[str]:
         rom_files: list[str] = []
 
-        for path, _, files in os.walk(f"{roms_path}/{rom}"):
-            for f in self._exclude_files(files, "multi_parts"):
-                rom_files.append(f"{Path(path, f)}".replace(f"{roms_path}/{rom}/", ""))
+        # Check if rom is a multi-part rom
+        if os.path.isdir(f"{roms_path}/{rom}"):
+            multi_files = os.listdir(f"{roms_path}/{rom}")
+            for file in multi_files:
+                with open(Path(roms_path, rom, file), "rb") as f:
+                    data = f.read()
```

Comment:
A good alternative to reading the entire file in memory can be found in [this answer](https://stackoverflow.com/a/1131238).

This will be important, specially for bigger files, to avoid using a lot of system memory while hashes are being calculated. Let's note that, with the approach provided in the linked snippet, we should still open and read the file a single time, updating all three hash objects on each read chunk.